spotxBidAdapter.js

```diff
@@ -0,0 +1,363 @@
+      } else {
+        bid.params.video['content_width'] = window.document.getElementById(videoSlotDivId).querySelectorAll('video')[0].offsetWidth;
+      }
+    }
+
+    if (utils.getBidIdParameter('content_height', bid.params.video) == '') {
+      if (window.document.getElementById(videoSlotDivId).querySelectorAll('video').length == 0) {
+        bid.params.video['content_height'] = window.document.getElementById(videoSlotDivId).offsetHeight;
+      } else {
+        bid.params.video['content_height'] = window.document.getElementById(videoSlotDivId).querySelectorAll('video')[0].offsetHeight;
+      }
+    }
+
+    utils.logWarn(ORTB_VERSION);
+
+    return true;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   * from Prebid.js: buildRequests - Takes an array of valid bid requests, all of which are guaranteed to have passed the isBidRequestValid() test.
+   *
+   * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(bidRequests, bidderRequest) {
+    const loc = utils.getTopWindowLocation();
+    const page = loc.href;
+    const isPageSecure = (loc.protocol === 'https:') ? 1 : 0;
+
+    let siteId = '';
+    let bid = bidderRequest.bids[0];
+    let channelId = bid.params.video.channel_id;
+    let pubcid = null;
+
+    const contentWidth = utils.getBidIdParameter('content_width', bid.params.video);
+    const contentHeight = utils.getBidIdParameter('content_height', bid.params.video);
+
+    const spotxImps = bidRequests.map(function(bid) {
+      const secure = isPageSecure || (utils.getBidIdParameter('secure', bid.params) ? 1 : 0);
+
+      // Mandatory
+      const adMute = utils.getBidIdParameter('ad_mute', bid.params.video) != '' ? 0 + Boolean(utils.getBidIdParameter('ad_mute', bid.params.video)) : 0;
+      const hideSkin = utils.getBidIdParameter('hide_skin', bid.params.video) != '' ? 0 + Boolean(utils.getBidIdParameter('hide_skin', bid.params.video)) : 0;
+
+      const ext = {
+        player_width: contentWidth,
+        player_height: contentHeight,
+        sdk_name: 'Prebid 1+',
+        ad_mute: adMute,
+        ad_unit: 'outstream',
+        hide_skin: hideSkin,
+        content_page_url: page,
+        versionOrtb: ORTB_VERSION,
+        bidId: bid.bidId,
+        videoSlot: bid.params.video.video_slot,
+        outstream_static: bid.params.video.outstream_static ? bid.params.video.outstream_static : false
+      };
+
+      if (utils.getBidIdParameter('ad_volume', bid.params.video) != '') {
+        ext.ad_volume = utils.getBidIdParameter('ad_volume', bid.params.video);
+      }
+
+      const mimes = utils.getBidIdParameter('mimes', bid.params.video) != '' ? utils.getBidIdParameter('mimes', bid.params.video) : ['application/javascript', 'video/mp4', 'video/webm'];
```

Comment:
Cant this just be utils.getBidIdParameter('mimes', bid.params.video) || ['application/javascript', 'video/mp4', 'video/webm'];